File: app/backend/schemas/community.py
from pydantic import BaseModel, Field


class CommunityPostCreate(BaseModel):
    category: str = Field(min_length=1, max_length=30)
    title: str = Field(min_length=1, max_length=200)
    content: str = Field(min_length=1, max_length=5000)


class CommunityCommentCreate(BaseModel):
    content: str = Field(min_length=1, max_length=1000)


class CommunityCommentResponse(BaseModel):
    comment_id: str
    customer_id: str
    company_name: str
    content: str
    created_at: float


class CommunityPostResponse(BaseModel):
    post_id: str
    customer_id: str
    company_name: str
    category: str
    title: str
    content: str
    created_at: float
    comment_count: int = 0


class CommunityPostDetailResponse(CommunityPostResponse):
    comments: list[CommunityCommentResponse] = []

File: web/lib/api/community.ts
import { apiFetch } from "./client";

export type CommunityPost = {
  post_id: string;
  customer_id: string;
  company_name: string;
  category: string;
  title: string;
  content: string;
  created_at: number;
  comment_count: number;
};

export type CommunityPostCreate = {
  category: string;
  title: string;
  content: string;
};

export type CommunityComment = {
  comment_id: string;
  customer_id: string;
  company_name: string;
  content: string;
  created_at: number;
};

export type CommunityPostDetail = CommunityPost & {
  comments: CommunityComment[];
};

export async function getCommunityPosts(): Promise<CommunityPost[]> {
  return apiFetch("/api/v1/community/posts");
}

export async function getCommunityPost(
  postId: string,
): Promise<CommunityPostDetail> {
  return apiFetch(`/api/v1/community/posts/${postId}`);
}

export async function createCommunityPost(
  input: CommunityPostCreate,
): Promise<CommunityPost> {
  return apiFetch("/api/v1/community/posts", {
    method: "POST",
    body: JSON.stringify(input),
  });
}

export async function createCommunityComment(
  postId: string,
  content: string,
): Promise<CommunityComment> {
  return apiFetch(`/api/v1/community/posts/${postId}/comments`, {
    method: "POST",
    body: JSON.stringify({ content }),
  });
}
